helm/data_repository.go

```diff
@@ -75,83 +74,63 @@ func dataRepository() *schema.Resource {
 
 func dataRepositoryRead(d *schema.ResourceData, meta interface{}) error {
 	m := meta.(*Meta)
+	m.Lock()
+	defer m.Unlock()
 
 	name := d.Get("name").(string)
-	err := addRepository(m,
-		name,
-		d.Get("url").(string),
-		m.Settings.Home,
-		d.Get("cert_file").(string),
-		d.Get("key_file").(string),
-		d.Get("ca_file").(string),
-		d.Get("username").(string),
-		d.Get("password").(string),
-	)
-
-	if err != nil {
-		return err
-	}
 
-	debug("%q has been added to your repositories\n", name)
+	//var entry *repo.Entry
+	var file *repo.File
 
-	r, err := getRepository(d, m)
-	if err != nil {
-		return err
-	}
-
-	return setIDAndMetadataFromRepository(d, r)
-}
-
-func getRepository(d *schema.ResourceData, m *Meta) (*repo.Entry, error) {
-	name := d.Get("name").(string)
+	if fileExists(m.Settings.RepositoryConfig) {
+		var err error
+		if file, err = repo.LoadFile(m.Settings.RepositoryConfig); err != nil {
+			return err
+		}
 
-	f, err := repo.LoadRepositoriesFile(m.Settings.Home.RepositoryFile())
-	if err != nil {
-		return nil, err
+	} else {
+		file = repo.NewFile()
 	}
 
-	for _, r := range f.Repositories {
-		if r.Name == name {
-			return r, nil
+	entry := file.Get(name)
+
+	// Not sure I agree with the logic here. Should a data source really update an underlying resource every time its called?
```

Comment:
Not sure about this behavior, and if it completely makes sense. Would like feed back.